linalg/petsc.cpp

```diff
@@ -178,10 +198,26 @@ void MFEMInitializePetsc(int *argc,char*** argv)
 void MFEMInitializePetsc(int *argc,char ***argv,const char rc_file[],
                          const char help[])
 {
+   // Tell PETSc to use the same CUDA or HIP device as MFEM:
    if (mfem::Device::Allows(mfem::Backend::CUDA_MASK))
    {
-      // Tell PETSc to use the same CUDA device as MFEM:
-      ierr = PetscOptionsSetValue(NULL,"-cuda_device",
+#if PETSC_VERSION_LT(3,17,0)
+      const char *opts = "-cuda_device";
+#else
+      const char *opts = "-device_select_cuda";
+#endif
+      ierr = PetscOptionsSetValue(NULL,opts,
+                                  to_string(mfem::Device::GetId()).c_str());
+      MFEM_VERIFY(!ierr,"Unable to set initial option value to PETSc");
+   }
+   if (mfem::Device::Allows(mfem::Backend::HIP_MASK))
+   {
+#if PETSC_VERSION_LT(3,17,0)
+      const char *opts = "-hip_device";
+#else
+      const char *opts = "-device_select_hip";
+#endif
```

Comment:
What is the backward compatibility policy/supported version range with HIP? I see `INSTALL` says `PETSc >= 3.15.0  (PETSc built with CUDA)`.

Reply:
I don't know @balay ?

Reply:
wrt HIP versions?  Right now we have minversion set to `4.1`, however we test only with hip-5.4.3. [looking at migrating to 5.6.0].

But generally older versions are supported - as in if we find build failures that are easy to fix - we would. [and if we are unable to fix - we can re-anchor the minversion].

Reply:
@balay, I think the question was about the versions of PETSc that can be used with HIP (and work with this branch).

In MFEM's `INSTALL` file, I added "PETSc >= 3.19.0  (PETSc built with HIP, older versions may work too)", however, this text can be adjusted if needed. I tested with PETSc 3.19.5 and assumed that 3.19.0 will work too.